packages/dx-react-scheduler/src/plugins/day-view.tsx

```diff
@@ -20,90 +21,143 @@ import {
 
-    this.timeTable = { current: null };
-    this.layout = React.createRef();
-    this.layoutHeader = React.createRef();
-    this.timeTableRef = this.timeTableRef.bind(this);
+  state = {
+    timeTableRef: null,
+  };
 
-    this.sidebarPlaceholder = () => <TemplatePlaceholder name="sidebar" />;
-    this.dayScalePlaceholder = () => <TemplatePlaceholder name="navbar" />;
-    this.dayScaleEmptyCellPlaceholder = () => <TemplatePlaceholder name="dayScaleEmptyCell" />;
-    this.timeTablePlaceholder = () => <TemplatePlaceholder name="main" />;
-    this.appointmentPlaceholder = params => <TemplatePlaceholder name="appointment" params={params} />;
-    this.cellPlaceholder = params => <TemplatePlaceholder name="cell" params={params} />;
+  static defaultProps: Partial<VerticalViewProps> = {
+    name: 'Day',
+    startDayHour: 0,
+    endDayHour: 24,
+    cellDuration: 30,
+    intervalCount: 1,
+  };
 
-    const {
-      name: viewName,
-      startDayHour,
-      endDayHour,
-      cellDuration,
-      intervalCount,
-    } = props;
+  static components: PluginComponents = {
+    layoutComponent: 'Layout',
+    appointmentLayerComponent: 'AppointmentLayer',
+    dayScaleEmptyCellComponent: 'DayScaleEmptyCell',
+    timeScaleLayoutComponent: 'TimeScaleLayout',
+    timeScaleCellComponent: 'TimeScaleCell',
+    timeScaleRowComponent: 'TimeScaleRow',
+    dayScaleLayoutComponent: 'DayScaleLayout',
+    dayScaleCellComponent: 'DayScaleCell',
+    dayScaleRowComponent: 'DayScaleRow',
+    timeTableLayoutComponent: 'TimeTableLayout',
+    timeTableCellComponent: 'TimeTableCell',
+    timeTableRowComponent: 'TimeTableRow',
+  };
 
-    const viewCellsDataComputed = ({
-      currentDate,
-    }) => viewCellsDataCore(
+  viewCellsDataBaseComputed = ({
+    currentDate,
+  }) => {
+    const { intervalCount, startDayHour, endDayHour, cellDuration } = this.props;
+    return viewCellsDataCore(
       currentDate, undefined,
       intervalCount, [],
-      startDayHour, endDayHour, cellDuration,
+      startDayHour!, endDayHour!, cellDuration!,
       Date.now(),
     );
+  }
 
-    const timeTableElementComputed = () => this.timeTable;
-    const layoutElementComputed = () => this.layout;
-    const layoutHeaderElementComputed = () => this.layoutHeader;
+  timeTableElementComputed = () => this.timeTable;
+  layoutElementComputed = () => this.layout;
+  layoutHeaderElementComputed = () => this.layoutHeader;
 
-    this.startViewDateComputed = getters => computed(
-      getters, viewName, startViewDateBaseComputed, getters.startViewDate,
+  layoutHeaderElement = (getters: Getters) => {
+    const { name: viewName } = this.props;
+    return computed(
+      getters, viewName!, this.layoutHeaderElementComputed, getters.layoutHeaderElement,
     );
-    this.endViewDateComputed = getters => computed(
-      getters, viewName, endViewDateBaseComputed, getters.endViewDate,
+  }
+
+  layoutElement = (getters: Getters) => {
```

Comment:
It should be

```jsx
layoutElement: ComputedFn = (getters) => { ... }
```

because `layoutElement` is actually a `computed` property value.